### DISCUSSION TITLE
Cable adaptors from PF to PowerUp and viceversa

### DISCUSSION THREAD

**Discussion by Salva-OA** [pybricks/support#490] [Q&A]
Hi,

Using this kind of adaptors
https://pv-productions.com/product/power-functions-to-powered-up-adapter/

and based on this info https://www.philohome.com/wedo2reverse/connect.htm

It is possible to plug a PF dc motor (4 pins) into the powerup port (6 pins).

Can we have a class for controlling wedo1 sensors (distance, tilt) plus all the range of power function motors directly?.

Thanks for your help

**Comment by dlech** [comment]
> It is possible to plug a PF dc motor (4 pins) into the powerup port (6 pins).

Yes, assuming the adapters use one of the wrings listed on Philo's page, they will already work with Pybricks. If they work with the official LEGO software already, then this is likely the case.

> Can we have a class for controlling wedo1 sensors (distance, tilt) plus all the range of power function motors directly?.

I'm not sure how the sensors work, but it seems unlikely that we would ever support them. The existing `DCMotor` class will work for all of the motors.
